### DISCUSSION TITLE
Quarto not rendering HTML coded math formula--bug, not bug, workaround?

### DISCUSSION THREAD

**Discussion by IceManTBM3** [quarto-dev/quarto-cli#4997] [Q&A] [answered]
I'm a content writer working with scientists and developers. so I'll disclose that up front.

We migrated from Visual Studio and Markdown to R Studio and Quarto, most went well exception if several mathbox formulas don't render. Is this a bug, or is there something I need to do it make it  work?

Here is one that does not work
$$\mbox{FMV}_{GAP}^{implied}  = 100* (\frac{\mbox{Implied_Interest_Rate}}{\mbox{FMV_Interest_Rate}} - 1)$$

other like this work fine:
$$\mbox{marginal effect} = (\exp(\beta) -1 )\times X$$

Attached is how the old set up rendered the exact same (top) formula and how it renders wrong in Quarto

![math box error](https://user-images.githubusercontent.com/129208946/228315642-5dce7ba1-e5c1-49f5-8f2e-3d0be6e5d207.PNG)
![long-FMV_eq1](https://user-images.githubusercontent.com/129208946/228315646-eec1e403-b447-41b3-a48c-09f7cc05661c.PNG)

**Comment by cscheid** [comment]
Can you share a full document that exhibits the behavior? The solution will be different depending on the rest of your document (in particular, the front matter, format, etc.)

**Reply by IceManTBM3** [reply]
Hey Carlos, looks like I was using "_" and it says as you can see above it says you can only use in Math Mode, which I believe is what $$\mbox is.... if I pull the "_"s out in ", semi works

We can only attach png from our intranet, but here is the section of the code.... but I think I figured it out that the "_" in "(\frac{\mbox{Implied_Interest_Rate}}{\mbox{FMV_Interest_Rate}}" it works/mostly works.... I can send you the chunk of code in text if need be, but try the formula above in a qmd file in R Studio

Funny I did not get an error when playing in R and Quarto and it rendered correctly in the R Studio Visual mode but would not in Preview or rendered HTML

**Reply by cscheid** [reply]
Got it. Unfortunately, we can't help further without a full .qmd file that exhibits the issue.

**Reply by IceManTBM3** [reply]
Ok thx, bummer also funny in my last comms to you when I say "_" sometimes the underscore disappears on post, yet is there when I edit the comm... weirdness

**Comment by IceManTBM3** [comment]
...well and it looks like I was using an illegal character ("_") though it says it's allowed in Math Mode which I believe is what I am using

**Comment by cderv** [comment, marked as answer]
> We migrated from Visual Studio and Markdown

Do you know what was used in your previous workflow to render Math in HTML ? This would help to know how to configure Quarto document. Let me explain why below

I can reproduce your issue with this simple document 

`````markdown
---
title: "Math"
format: html
---

$$
\mbox{FMV}_{GAP}^{implied}  = 100* (\frac{\mbox{Implied_Interest_Rate}}{\mbox{FMV_Interest_Rate}} - 1)
$$

`````

By default this is rendered using MathJax 3. But if I change the math method to be Mathml for example, then it works
````markdown
---
title: "Math"
format: html
html-math-method: mathml
---

$$
\mbox{FMV}_{GAP}^{implied}  = 100* (\frac{\mbox{Implied_Interest_Rate}}{\mbox{FMV_Interest_Rate}} - 1)
$$
````

It seems Mathjax does not know how to render `\mbox{` inside `\frac{` as this works 
````
$$
\mbox{FMV}_{GAP}^{implied}
$$
````

but this does not 
````
$$
\frac{\mbox{Implied_Interest_Rate}}{\mbox{FMV_Interest_Rate}}
$$
````

It seems like a Mathjax bug, or regression or else as I can make your equation works if you use version 2.7.2 

````markdown
---
title: "Math"
format: html
html-math-method: 
  method: mathjax
  url: https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.2/MathJax.js?config=TeX-AMS-MML_HTMLorMML
---

$$\mbox{FMV}_{GAP}^{implied}  = 100* (\frac{\mbox{Implied_Interest_Rate}}{\mbox{FMV_Interest_Rate}} - 1)$$

````

About Math in Quarto https://quarto.org/docs/output-formats/html-basics.html#latex-equations

Hope this helps. We can't do much on this unfortunately.

**Reply by IceManTBM3** [reply]
Thanks so much. To answer your Q, I don't really know (I'm an editor guy), but this info does shed light on things. I'll talk to some of the dev folk I work with. Thank you again for trying out and giving us more options. 

FWIW, I got it to work by pulling the "_" JUST from the economic names like "Implied_Interest_Rate" and not the ones that create an inline spit....

**Reply by cderv** [reply]
I missed the hint on underscore. You can then just escape the underscore with `\`

````
$$\mbox{FMV}_{GAP}^{implied}  = 100* (\frac{\mbox{Implied\_Interest\_Rate}}{\mbox{FMV\_Interest\_Rate}} - 1)$$
````

This should show fine in the output, (but not fine in preview if uses mathjax 2 maybe...) - I don't why Mathajx 3 has this issue. Mentions of this in https://github.com/mathjax/MathJax/issues/329 too

**Reply by IceManTBM3** [reply]
Doh! and I knew that, thanks for the astute observation and the reminder. As is I think it looks much, much better sans the "_". 

thx again so much

**Comment by stefanbschneider** [comment]
For me, the solution was much simpler: I just had to escape the underscores with a `\`. Then, everything worked fine.
